app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -116,62 +127,100 @@ public void render(Chart chart, Map<String, ObsValue> latestObservations,
             for (ChartSection tileGroup : chart.tileGroups) {
                 List<Tile> tileRow = new ArrayList<>();
                 for (ChartItem item : tileGroup.items) {
-                    ObsValue[] obsValues = new ObsValue[item.conceptUuids.length];
-                    for (int i = 0; i < obsValues.length; i++) {
-                        obsValues[i] = latestObservations.get(item.conceptUuids[i]);
+                    ObsPoint[] points = new ObsPoint[item.conceptUuids.length];
+                    for (int i = 0; i < points.length; i++) {
+                        Obs obs = latestObservations.get(item.conceptUuids[i]);
+                        if (obs != null) {
+                            points[i] = obs.getObsPoint();
+                        }
+                    }
+                    tileRow.add(new Tile(item, points));
+                    if (!item.script.trim().isEmpty()) {
+                        mConceptsToDump.addAll(Arrays.asList(item.conceptUuids));
```

Comment:
Currently all tile and row concepts are dumped to JSON right? Should we limit it to dump only those with something in the "script" field? Would there be any reason to dump a concept to JSON if there is nothing in the script field?